quic/s2n-quic/src/connection/handle.rs

```diff
@@ -169,6 +169,14 @@ macro_rules! impl_handle_api {
             self.0.application_protocol()
         }
 
+        /// Returns the application context the connection is using.
```

Comment:
We should probably clarify expected usage with this API. A link to the TLS provider method would be good. We should also say something about this consuming the context and calling it a second time will always return None so applications should store it somewhere.